store/sqlstore/retention_policy_store.go

```diff
@@ -0,0 +1,529 @@
+	LIMIT :Limit
+	OFFSET :Offset`
+	props := map[string]interface{}{"PolicyId": policyId, "Limit": limit, "Offset": offset}
+	_, err = s.GetReplica().Select(&channels, query, props)
+	return
+}
+
+func (s *SqlRetentionPolicyStore) AddChannels(policyId string, channelIds []string) error {
+	if len(channelIds) == 0 {
+		return nil
+	}
+	if err := s.checkChannelsExist(channelIds); err != nil {
+		return err
+	}
+	builder := s.getQueryBuilder().
+		Insert("RetentionPoliciesChannels").
+		Columns("policyId", "channelId")
+	for _, channelId := range channelIds {
+		builder = builder.Values(policyId, channelId)
+	}
+	_, err := builder.RunWith(s.GetMaster()).Exec()
+	if err != nil {
+		switch dbErr := err.(type) {
+		case *pq.Error:
+			if dbErr.Code == PGForeignKeyViolationErrorCode {
+				return store.NewErrNotFound("RetentionPolicy", policyId)
+			}
+		case *mysql.MySQLError:
+			if dbErr.Number == MySQLForeignKeyViolationErrorCode {
+				return store.NewErrNotFound("RetentionPolicy", policyId)
+			}
+		}
+	}
+	return err
+}
+
+func (s *SqlRetentionPolicyStore) RemoveChannels(policyId string, channelIds []string) error {
+	if len(channelIds) == 0 {
+		return nil
+	}
+	builder := s.getQueryBuilder().
+		Delete("RetentionPoliciesChannels").
+		Where(sq.And{
+			sq.Eq{"PolicyId": policyId},
+			inStrings("ChannelId", channelIds),
+		})
+	_, err := builder.RunWith(s.GetMaster()).Exec()
+	return err
+}
+
+func (s *SqlRetentionPolicyStore) GetTeams(policyId string, offset, limit int) (teams []*model.Team, err error) {
+	const query = `
+	SELECT Teams.* FROM RetentionPoliciesTeams
+	INNER JOIN Teams ON RetentionPoliciesTeams.TeamId = Teams.Id
+	WHERE RetentionPoliciesTeams.PolicyId = :PolicyId
+	ORDER BY Teams.DisplayName, Teams.Id
+	LIMIT :Limit
+	OFFSET :Offset`
+	props := map[string]interface{}{"PolicyId": policyId, "Limit": limit, "Offset": offset}
+	_, err = s.GetReplica().Select(&teams, query, props)
+	return
+}
+
+func (s *SqlRetentionPolicyStore) AddTeams(policyId string, teamIds []string) error {
+	if len(teamIds) == 0 {
+		return nil
+	}
+	if err := s.checkTeamsExist(teamIds); err != nil {
+		return err
+	}
+	builder := s.getQueryBuilder().
+		Insert("RetentionPoliciesTeams").
+		Columns("PolicyId", "TeamId")
+	for _, teamId := range teamIds {
+		builder = builder.Values(policyId, teamId)
+	}
+	_, err := builder.RunWith(s.GetMaster()).Exec()
+	return err
+}
+
+func (s *SqlRetentionPolicyStore) RemoveTeams(policyId string, teamIds []string) error {
+	if len(teamIds) == 0 {
+		return nil
+	}
+	builder := s.getQueryBuilder().
+		Delete("RetentionPoliciesTeams").
+		Where(sq.And{
+			sq.Eq{"PolicyId": policyId},
+			inStrings("TeamId", teamIds),
+		})
+	_, err := builder.RunWith(s.GetMaster()).Exec()
+	return err
+}
+
+// RemoveOrphanedRows removes entries from RetentionPoliciesChannels and RetentionPoliciesTeams
+// where a channel or team no longer exists.
+func (s *SqlRetentionPolicyStore) RemoveOrphanedRows(limit int) (deleted int, err error) {
```

Comment:
For another PR, right?

Reply:
Yes, this is going to be used in the wireup PR.